src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,100 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that
+/// a prompt is printed before capturing input.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let user_input = prompt("Please enter some text: ");
+///
+///     println!("You typed: {}", user_input);
```

Comment:
```suggestion
///     let name = prompt("Name: ");
///
///     println!("Your name: {}", name);
```
Have a better example, I wonder if we should show them to handle empty input.

Reply:
Changed the example, What do you mean by "show them to handle empty input"?
Do you mean something like this?
```
let name = prompt("Enter name: ");

if name.len() == 0 {
    println!("No input!");
} else {
    println!("Your name: {}", name);
}
```